projects/core/src/http/http-timeout/http-timeout.interceptor.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpEventType,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { NEVER, Observable, of, TimeoutError } from 'rxjs';
+import { catchError, startWith, switchMap, timeout } from 'rxjs/operators';
+import { OccConfig } from '../../occ/config/occ-config';
+import { WindowRef } from '../../window/window-ref';
+import { HTTP_TIMEOUT_CONFIG } from './http-timeout.config';
+
+/**
+ * It throws an error when a request takes longer than the specified time.
```

Comment:
I think the previous explanation was better here (ie. specifying gateway timeout and status code).

Reply:
I no longer return artificial status code nor the statusText, but only the Error object. See the [recent commit](https://github.com/SAP/spartacus/pull/16658/commits/ac8216e28aac30287476e2d6957b9802658bb7a2)